MolecularNodes/pref.py

```diff
@@ -43,25 +42,10 @@ class MOL_OT_install_dependencies(bpy.types.Operator):
     
     def execute(self, context):
         if not pkg.available():
-            import datetime
+            stderr=pkg.install(
+                pypi_mirror=bpy.context.scene.pypi_mirror,
+            )
             
-            # generate logfile
-            logfile_path = os.path.abspath(MolecularNodesAddon.logpath + 'side-packages-install.log')
-            logfile = open(logfile_path, 'a')
-            
-            logfile.write("-----------------------------------" + '\n')
-            logfile.write("Installer Started: " + str(datetime.datetime.now()) + '\n')
-            logfile.write("-----------------------------------" + '\n')
-            
-            pkg.install()
```

Comment:
This section included the date times, but also included the result of `pkg.install()` which wrote the terminal output to the log file. On mac you can't view the terminal without first starting Blender from the terminal, so a user can't see any proper error outputs if they haven't started from the command line. 

I think we should probably still be writing some kind of log file, especially as we add the requirements for more and more python packages to add more functionality, as it'll help with debugging in the future.

Reply:
Okay, that is a good point. But it looks like the error message is not correctly recorded, as reported in #108 and #111 for instances. I think there might be a proper solution by redirecting the stdin, stdout and stderr respectively to pkt_install() and writing them into logfile.

Reply:
yes this is true. The logging seemed to work on my machine but was problematic with others. I need someone with more python knowledge than me to potentially fix that. Maybe we can merge this PR regardless and then try and set up some logging in another.

Reply:
I think it should be fixed now. for Windows: https://gist.github.com/YaoYinYing/5ac064ad2c6e642ab151583252e6cfbe
and for Mac on M1: https://gist.github.com/YaoYinYing/77d746c3cee28a7766125c9df9d2df09

also for the error of Python header, an user guide link will appear at the Blender console: 

<img width="737" alt="image" src="https://user-images.githubusercontent.com/33014714/219921900-d9626ce6-5282-4c1e-942e-91ea4e88a655.png">